client/my-sites/plans-v2/details.tsx

```diff
@@ -68,6 +78,24 @@ const DetailsPage = ( { duration, productSlug, rootUrl, header }: DetailsPagePro
 		checkout( siteSlug, slug );
 	};
 
+	const onDurationChange = ( newDuration: Duration ) => {
+		if ( newDuration === duration ) {
+			return;
+		}
+
+		const newProductSlug =
```

Comment:
This feels a bit fragile to me, but I opted not to over-plan for the future.

Reply:
Yeah, I guess in the future we could move the logic into its own function and handle any irregular case in there.